snapcraft/plugins/kernel.py

```diff
@@ -81,25 +81,30 @@
     'arm64': 'Image.gz',
 }
 
-required_generic = ['DEVTMPFS', 'DEVTMPFS_MOUNT', 'TMPFS_POSIX_ACL', 'IPV6',
-                    'SYSVIPC', 'SYSVIPC_SYSCTL', 'VFAT_FS', 'NLS_CODEPAGE_437',
-                    'NLS_ISO8859_1']
-
-required_security = ['SECURITY', 'SECURITY_APPARMOR', 'SYN_COOKIES',
-                     'STRICT_DEVMEM', 'DEFAULT_SECURITY_APPARMOR', 'SECCOMP',
-                     'SECCOMP_FILTER', 'CC_STACKPROTECTOR',
-                     'CC_STACKPROTECTOR_STRONG', 'DEBUG_RODATA',
-                     'DEBUG_SET_MODULE_RONX']
-
-required_snappy = ['RD_LZMA', 'KEYS', 'ENCRYPTED_KEYS', 'SQUASHFS',
-                   'SQUASHFS_XATTR', 'SQUASHFS_XZ',
-                   'DEVPTS_MULTIPLE_INSTANCES']
-
-required_systemd = ['DEVTMPFS', 'CGROUPS', 'INOTIFY_USER', 'SIGNALFD',
-                    'TIMERFD', 'EPOLL', 'NET', 'SYSFS', 'PROC_FS', 'FHANDLE',
-                    'DMIID', 'BLK_DEV_BSG', 'NET_NS',
-                    'IPV6', 'AUTOFS4_FS',
-                    'TMPFS_POSIX_ACL', 'TMPFS_XATTR', 'SECCOMP']
+required_generic = {'DEVTMPFS': '', 'DEVTMPFS_MOUNT': '',
+                    'TMPFS_POSIX_ACL': '', 'IPV6': '', 'SYSVIPC': '',
+                    'SYSVIPC_SYSCTL': '', 'VFAT_FS': '',
+                    'NLS_CODEPAGE_437': '', 'NLS_ISO8859_1': ''}
+
+required_security = {'SECURITY': '', 'SECURITY_APPARMOR': '',
+                     'SYN_COOKIES': '', 'STRICT_DEVMEM': '',
+                     'DEFAULT_SECURITY_APPARMOR': '', 'SECCOMP': '',
+                     'SECCOMP_FILTER': '', 'CC_STACKPROTECTOR': '',
+                     'CC_STACKPROTECTOR_STRONG':
+                     '(4.1.x and later versions only)',
+                     'DEBUG_RODATA': '', 'DEBUG_SET_MODULE_RONX': ''}
+
+required_snappy = {'RD_LZMA': '', 'KEYS': '', 'ENCRYPTED_KEYS': '',
+                   'SQUASHFS': '', 'SQUASHFS_XATTR': '', 'SQUASHFS_XZ': '',
+                   'DEVPTS_MULTIPLE_INSTANCES':
+                   '(4.8.x and earlier versions only)'}
+
+required_systemd = {'DEVTMPFS': '', 'CGROUPS': '', 'INOTIFY_USER': '',
+                    'SIGNALFD': '', 'TIMERFD': '', 'EPOLL': '',
+                    'NET': '', 'SYSFS': '', 'PROC_FS': '', 'FHANDLE': '',
+                    'DMIID': '', 'BLK_DEV_BSG': '', 'NET_NS': '',
+                    'IPV6': '', 'AUTOFS4_FS': '',
+                    'TMPFS_POSIX_ACL': '', 'TMPFS_XATTR': '', 'SECCOMP': ''}
```

Comment:
Hey @piso77, I'm not sure that a dictionary is the right structure for this, because it's mostly empty.
The nicer message seems good to me, but if it only applies to two cases, adding two `if`s in the code is better.